### PR TITLE
Added pets, hero effects and kill effects

### PR DESCRIPTION
**Pull request by Shesmuu**
Added hero effects, for this, the ethereal gems particles from regular dota were used, and the color of these effects can be changed to colors of prismatic gems. Added kill effects tab, but so far there is 1 effect. Added the ability to choose a pet, a choice of 261 courier models

### CODE REVIEW

**Comment by darklordabc** [comment]
This is not completed yet, no need to review.

File: game/scripts/vscripts/common/cosmetic_abilities.lua [hunk 1]
```diff
@@ -1,104 +1,10262 @@
...
+	print( "Cosmetics precache start" )
+
+	for _, effect in pairs( self.heroEffects ) do
+		if effect.resource then
+			PrecacheResource( "particle_folder", effect.resource, context )
+		end
+	end
+
+	for _, c in pairs( self.petsData.couriers ) do
+		PrecacheModel( c.ground_radiant.model, context )
```
Comment (ark120202) [inline game/scripts/vscripts/common/cosmetic_abilities.lua]: That might cause performance issues, would be nice to measure loading times before and after this change, especially on weak clients

File: game/scripts/vscripts/common/cosmetic_abilities.lua [hunk 2]
```diff
@@ -1,104 +1,10262 @@
...
+
+	local ability = unit:AddAbility( keys.ability )
+
+	ability:SetLevel( 1 )
+	ability:SetHidden( false )
+
+	CustomGameEventManager:Send_ServerToAllClients( "cosmetics_reload_abilities", nil )
+end
+
+function Cosmetics.RemvoeAbility( keys )
```
Comment (ark120202) [inline game/scripts/vscripts/common/cosmetic_abilities.lua]:
```suggestion
function Cosmetics.RemoveAbility( keys )
```

File: game/scripts/vscripts/common/cosmetic_abilities.lua [hunk 3]
```diff
@@ -1,104 +1,10262 @@
...
+end
+
+function Cosmetics:OnNPCSpawned( keys )
+	local unit = EntIndexToHScript( keys.entindex )
+	local n = unit:GetUnitName()
+
+	if unit:IsRealHero() and not unit.cosmeticsLoaded then
+		local id = unit:GetPlayerID()
+		--[[
+		WebApi:Send(
```
Comment (ark120202) [inline game/scripts/vscripts/common/cosmetic_abilities.lua]: See https://github.com/dota2unofficial/overthrow2/commit/618da30b081d9666b553f83cece334b28202fee8

**Pull request merged**